src/main/scala/domain/User.scala

```diff
@@ -0,0 +1,12 @@
+package domain
+
+case class UserId(id: String)
+
+case class Password(password: String)
+
+case class User( id: UserId,
+                 name: String,
+                 password: Password,
```

Comment:
I don't think we need the password and salt to exist in the User class - it's a security risk, we'd never want to send that data back to the browser.